crates/fuel-data-parser/src/error.rs

```diff
@@ -0,0 +1,47 @@
+#![allow(dead_code)]
```

Comment:
Since we already define which types of algorithms we will use, to avoid delivering a bigger binary, maybe you could add extra stuff under `cfg` flags, what do you think?

Reply:
I dont think I can put cfg for each compr/ser algo as the latter are all part of an external crate and internally we just switch based on the user conifiguration. What I am saying here is that there arent any based on the ser or compression specific methods, but rather methods that match internally on the specified config. Or maybe you have some idea here ?

Reply:
It's just because since we chose to go with postcard, we could remove code related to other external serializers, but to avoid deleting, we can just use the `cfg`